src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -32,30 +35,40 @@ interface DropdownMenuClickProps<T> extends DropdownMenuBaseProps<T> {
 interface DropdownMenuContextProps<T> extends DropdownMenuBaseProps<T> {
   trigger: 'contextMenu';
   children: ReactNode;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  as?: ElementType | React.ComponentType<any>;
```

Comment:
You don't need the `React.ComponentType`, and `as` should be using the `E` generic.